### DISCUSSION TITLE
Using TuiDocMainModule in multiple modules

### DISCUSSION THREAD

**Discussion by Teebo** [taiga-family/taiga-ui#1077] [Q&A] [answered]
I am using the `TuiDocMainModule` in 3 modules.

One of them is `app.module.ts` which declares a module that makes use of the **auto-complete input**
In the `app.component.html` I am also checking the route and deciding if I use the Angular `router-outlet` or `<tui-doc-main></tui-doc-main>`

```
<tui-root>
  <router-outlet *ngIf="(isRouteWithoutDocModule$ | async); else portal"></router-outlet>

  <ng-template #portal>
    <tui-doc-main></tui-doc-main>
  </ng-template>
</tui-root>

```

If I do not import `TuiDocMainModule` in the `app.module.ts` the auto-complete input works as expected but if I do import  `TuiDocMainModule`, my other modules that are using the documentation engine (also needing the `TuiDocMainModule`) fail to render and there is no error thrown.

If I remove the `TuiDocMainModule` , I get an error in the `app.module.ts ` because` <tui-doc-main></tui-doc-main>` will not exist in the app module to be used in the ` app.component.htm`. But then the modules that are using `TuiDocMainModule` do render.

How can I resolve this issue? I understand it could not be TaigaUI but rather a  dependency conflict.

**Comment by waterplea** [comment, marked as answer]
`tui-root` is included in `tui-doc-main`. If you can make a github repo I could take a look.
